### DISCUSSION TITLE
General discussion for questions, chatting, reports and requests!

### DISCUSSION THREAD

**Discussion by Aris-t2** [Aris-t2/NoiaButtons#4] [General]
**Do not open new issue threads for general talk or questions. Use this post.**

**Comment by pro100Hub** [comment]
@Aris-t2 
Hello.
I use `noiaui_light_tabs_below_navigation_toolbar` and `noiaui_light_tabs_below_navigation_toolbar_fx65`, but after update to FF 71 they again moved up :( Tell please, you can fix this?

Regards.

**Comment by Aris-t2** [comment]
I will look into it. Can you test this till then?

```css
#TabsToolbar {
  display: block !important;
}
```

**Comment by pro100Hub** [comment]
It works. Thank you so much, Aris!

Result in `noiaui_light_tabs_below_navigation_toolbar_fx65` (`display: block !important;`) ↓
```
#TabsToolbar {
  display: block !important;
  position: absolute !important;
  bottom: 0 !important;
  width: 100vw !important;  
}
```
In order for the navigation arrows to be black, not white, added to `noiaui_light_tabs_below_navigation_toolbar` (`color: unset !important;`)

```
#TabsToolbar:not(:-moz-lwtheme){
  -moz-appearance: none !important;
  background-image: var(--tabs_toolbar_color_tabs_not_on_top) !important;
  color: unset !important;
}
```

@Aris-t2 , if you have time, please look at these files: `noiaui_light_tabs_below_navigation_toolbar` and `noiaui_light_tabs_below_navigation_toolbar_fx65`. Maybe they require optimization for the new version of FF 71 and there may be old and not relevant code?
You can use my files ↓
[(css) noiaui_light_tabs_below_navigation_toolbar.zip](https://github.com/Aris-t2/NoiaButtons/files/3921886/css.noiaui_light_tabs_below_navigation_toolbar.zip)

**Comment by Aris-t2** [comment]
I posted updated code and a few fixes.

**Comment by pro100Hub** [comment]
Thanks, @Aris-t2 :+1: 
It seems that `noiaui_light_tabs_below_navigation_toolbar` is missing this code ↓ 
```
/* fix Windows 7s default tab background (remove dark purple bg color)*/
@media (-moz-os-version: windows-win7) {
  @media (-moz-windows-default-theme) {
    .tabbrowser-tab:not([selected]):not(:hover):not(:-moz-lwtheme) {
      background-color: #e3e4e6 !important;
    }
	.tabbrowser-tab:not([selected]):hover:not(:-moz-lwtheme) {
      background-color: #cccdcf !important;
    }

  }
}

/* toolbar colors */
#main-window:not([tabsintitlebar]) #toolbar-menubar:not(:-moz-lwtheme) {
  background-image: var(--colored_menubar_background_image) !important;
}

#nav-bar:not(:-moz-lwtheme) {
  -moz-appearance: none !important;
  background: var(--general_toolbar_color_navbar, inherit) !important;
}

toolbar:not(#TabsToolbar):not(#toolbar-menubar):not(#nav-bar):not(:-moz-lwtheme) {
  -moz-appearance: none !important;
  background: var(--general_toolbar_color_toolbars, inherit) !important;
}
#TabsToolbar:not(:-moz-lwtheme){
  -moz-appearance: none !important;
  background-image: var(--tabs_toolbar_color_tabs_not_on_top) !important;
  color: unset !important;
}

/* remove color overlay for lw-themes */
#main-window[style*='--lwt-header-image'] :-moz-any(#nav-bar,#PersonalToolbar,#TabsToolbar):-moz-lwtheme{
  background: unset !important;
}

/* adjust compact themes background color */
#main-window:not([style*='--lwt-header-image']) #TabsToolbar:-moz-lwtheme {
  -moz-appearance: none !important;
  background-image: linear-gradient(var(--toolbar-bgcolor),var(--toolbar-bgcolor)) !important;
}

#main-window:not(:-moz-lwtheme) #TabsToolbar :-moz-any(.tabs-newtab-button,.scrollbutton-up,.scrollbutton-down),
#main-window:not(:-moz-lwtheme) #TabsToolbar > toolbarbutton,
#main-window:not([style*='--lwt-header-image']):-moz-lwtheme:-moz-lwtheme-darktext #TabsToolbar :-moz-any(.tabs-newtab-button,.scrollbutton-up,.scrollbutton-down),
#main-window:not([style*='--lwt-header-image']):-moz-lwtheme:-moz-lwtheme-darktext #TabsToolbar > toolbarbutton {
  fill: var(--classic_squared_tabs_new_tab_icon_color) !important;
  color: var(--classic_squared_tabs_new_tab_icon_color) !important;
}
```
This code is required to fill in all additional toolbars.

**Comment by Aris-t2** [comment]
Are you sure, you are using the correct file in your userChrome.css? Looked fine for me in my tests.

Depending on the Firefox version you have to "enable" the correct file only.

noiaui_light_tabs_below_navigation_toolbar.css       <--- Fx60-64
or
noiaui_light_tabs_below_navigation_toolbar_fx65.css    <---- Fx65-71
or
noiaui_light_tabs_below_navigation_toolbar_fx72.css    <---- Fx72+

**Comment by pro100Hub** [comment]
@Aris-t2 
Reply to comment [#700214794](https://github.com/Aris-t2/CustomCSSforFx/issues/308#issuecomment-700214794) and [#700855569](https://github.com/Aris-t2/CustomCSSforFx/issues/308#issuecomment-700855569)

I understood what was the problem of overlapping min/max/close buttons.
In the file `noiaui_light_tabs_below_navigation_toolbar_fx65.css` commented out everything that is in the _Windows 7 extra tweaks_ line. Now the custom background works and does not overlap the buttons.

Aris, maybe this file `noiaui_light_tabs_below_navigation_toolbar_fx65.css` already has some irrelevant code for FF 81? Is it possible to optimize the code for FF 81?

I am using files: 
```
noiaui_light.css
noiaui_light_tabs_below_navigation_toolbar.css)
noiaui_light_tabs_below_navigation_toolbar_fx65.css)
```
[Files-css.zip](https://github.com/Aris-t2/NoiaButtons/files/5332831/Files-css.zip)

```

**Comment by githwithit** [comment]
I just updated my FirefoxESR/CustomCSSforFx from 68-2.5.3 to 78-3.1.7 and am having an issue I can not fix.

Now with FF78 and 3.1.7, when I highlight the URL text it changes from black/grey to blue and the background stays white.

With FF68 and 2.5.3 the text flipped to white and the background became blue when highlighted.

I confirmed that native FF78 flips to white text on blue for highlighted URLs.

How do I go back to the older behavior with white text on a blue background when highlighting a URL? TIA!

**Comment by Aris-t2** [comment]
@pro100Hub 
Post screenshots.

@githwithit 
Use on of these code snippets (present in CustomCSSforFx) and change the colors to whatever you like:

selection_color_for_bright_bg.css
```css
#nav-bar:-moz-lwtheme #urlbar ::-moz-selection,
#nav-bar:not(:-moz-lwtheme) #urlbar ::-moz-selection {
  background-color: Highlight !important;
  color: HighlightText !important;
}
```

selection_color_for_dark_bg.css
```css
#nav-bar:-moz-lwtheme #urlbar ::-moz-selection,
#nav-bar:not(:-moz-lwtheme) #urlbar ::-moz-selection {
  background-color: HighlightText !important;
  color: Highlight !important;
}
```

**Comment by pro100Hub** [comment]
@Aris-t2 
Sorry, what screenshot do you need?

**Comment by Aris-t2** [comment]
I need to know what exactly causes issues for you. The "complete" package offers "tabs not on top" files for multiple Firefox versions and has to be used like this:

![image](https://user-images.githubusercontent.com/6013569/95441785-d9294800-095a-11eb-95f8-00fe7059e89e.png)

The result in Firefox 83 is shown on this screenshot:
![image](https://user-images.githubusercontent.com/6013569/95441538-894a8100-095a-11eb-832c-bef65399cea6.png)


I know there was an issue with Win 7 AeroGlass on your system. You posted a small screenshot of caption buttons area in the CustomCSSforFx thread.

**Comment by pro100Hub** [comment]
I have commented out the  "_Windows 7 extra tweaks_" line in `noiaui_light_tabs_below_navigation_toolbar_fx65.css`
![css](https://user-images.githubusercontent.com/10475278/95571027-3f889600-0a41-11eb-9d50-6b12026fba48.png)

Filling the background with color around buttons now works fine.
![browser](https://user-images.githubusercontent.com/10475278/95571334-b160df80-0a41-11eb-9b8d-8eb6a7b68187.png)

**Comment by pro100Hub** [comment]
@Aris-t2 , please update `noiaui_light_tabs_below_navigation_toolbar` for FF 89.

**Comment by Aris-t2** [comment]
@pro100Hub 

https://github.com/Aris-t2/NoiaButtons/issues/3

**Comment by Tallion99** [comment]
Hi, I am having an issue getting the buttons to kick in.
I was at version 68 but needed to update, which it did to V72 and then it updated to the latest V91.
At this point the buttons continued to work.
I had a problem with the text in my URL bar so i did an uninstall and reinstalled V81. 

I tried copying over the chrome folder to the new profile but it did not work.
My work pc is V89 so i updated to V89 and copied over the chrome folder but that also did not work.
I removed all chrome folders and copied over your V2.0.0.0 but no luck.
I know it should just work but have now spent a few hours trying to figure it with no luck.
Seems like it is just ignored...
Any ideas?

Thanks, T

Edit: I had to change toolkit.legacyUserProfileCustomizations.stylesheets to True. It now working!

**Comment by Aris-t2** [comment]
Did you enable the about:config switch, so Firefox knows it is allowed to use custom CSS?

`about:config` > `toolkit.legacyUserProfileCustomizations.stylesheets` > `true`

**Comment by Tallion99** [comment]
Yes, just figured it out. I knew it was something along those lines but couldn't recall it! Thanks for the project! Have had these buttons for so long!

**Comment by pro100Hub** [comment]
@Aris-t2 
In FF 108, when using "[noiaui_light_tabs_below_navigation_toolbar.css](https://github.com/Aris-t2/NoiaButtons/blob/master/noiabuttons/css/noiaui_light_tabs_below_navigation_toolbar.css)" some navigation buttons are hidden, namely: `AllTabs-Button` and `New-Tab-Button` are not displayed (hidden).
If change the value in line No. 162 and 166 instead of `100vw` assign `100%`, then the buttons appear, but is this the right solution?

![Screenshot 2022-12-14 at 19-26-03 NoiaButtons_noiabuttons at master · Aris-t2_NoiaButtons](https://user-images.githubusercontent.com/10475278/207621624-c4b6dafd-dfdc-4588-ba7b-8ab61a4c591f.png)

Please update file on github :pray:

**Reply by Aris-t2** [reply]
Yes, changing  100vw to 100% is the right solution. I used this fix in CustomCSSforFx a while ago.

**Comment by pro100Hub** [comment]
@Aris-t2 
FF 111
When using `noiaui_light_tabs_below_navigation_toolbar.css` browser buttons (min, collapse to window, close), which are in the upper right corner, are hidden by half. Fix please.
The reason is `/* use caption buttons offered for lw-themes */`

The display of icons also disappears when using `Icons for default browser buttons`.
![изображение](https://user-images.githubusercontent.com/10475278/225533296-b44b909f-29ab-4fea-80a8-fb84164892e2.png)

~~And please tell how to completely move the `nav-bar` panel to the header and so that it does not hide when using a full screen through F11 on the keyboard?~~
![firefox_llSpSxhu8l](https://user-images.githubusercontent.com/10475278/225532567-14eab8a4-758c-41ad-a2a8-364acf1fa520.jpg)
**UPD:** 
```
/* Moves the address bar to Header. */
#main-window #navigator-toolbox #nav-bar:not([inFullscreen]) {
  margin-top: -26px !important;
  margin-inline-end: 140px !important;
}
```

**Reply by Aris-t2** [reply]
Update version went online today.

**Comment by 55trucker** [comment]
Aris:

using your script as a starting point for the css file, how does one enter additional script to include additional changes to the chrome file for making more changes such as adding a line separator in between the menu bar & the toolbar & as well between the toolbar & the tab bar?...this would recreate the original Noia theme visual.
I currently use Pale Moon as my browser but many websites no longer provide support for Pale Moon so I will go back to Firefox if I can reproduce the Noia theme for the current  Firefox browser theu the chrome file.....thx

**Reply by Aris-t2** [reply]
You could attach code at the bottom of userChrome.css for example.
Or add `@import "./my_userChrome.css"; `, create a new `my_userChrome.css` and add your code there.

**Reply by 55trucker** [reply]
Doing some investigating on userchrome one can add a supplemental entry named *whatever* .css & insert an entry into the very beginning of userChrome.css pointing to the supplemental file. If one does this how many *pointers* can be included in the original userChrome file before that function no longer works? 

As well, what would be the correct script to input into chrome to make the horizontal separators reappear between the menu bar & the navigator bar?..plus the nav bar & the tabs bar?....this is all related to the Noia theme.

I can't upload an image here to display to you the horizontal separators I am referring to.......

**Comment by 55trucker** [comment]
Hi there Aris:

how difficult would it be for you to enter a script into your Noia buttons project to add the horizontal dividers between the top menu & the navigator bar ....& between the navigator bar & the tabs bar that span the entire screen width?

I am trying to get this into place & using script written by others isn't functioning.

It's a case of *one doesn't like the other*..I can have the horizontal bars & not your Noia script or your Noia script & no bars. There's a conflict between the two scripts & they both won't appear.

Here is the script I am entering & it is not being accepted when I have your Noia Buttons script entered. If I remove your work then the bar script does appear.

Even if I use the method of a secondary userchrome file & place a pointer (import) in/at the top of the primary userchrome file pointing to the secondary file it still doesn't work


thx,


Tony


#navigator-toolbox #nav-bar {
border-top: 1px solid black !important;
margin-bottom: 0px !important;
}

#navigator-toolbox #PersonalToolbar {
border-top: 1px solid black !important;
margin-bottom: 0px !important;
}

#navigator-toolbox #TabsToolbar {
border-top: 1px solid black !important;
margin-bottom: 0px !important;
}


     On Monday, October 23, 2023 at 12:12:52 p.m. EDT, Aris ***@***.***> wrote:  
 
 


You could attach code at the bottom of userChrome.css for example.
Or add @import "./my_userChrome.css"; , create a new my_userChrome.css and add your code there.

**Reply by Aris-t2** [reply]
Give 3.0.1 release a try: https://github.com/Aris-t2/NoiaButtons/releases

**Comment by 55trucker** [comment]
You are a Wizard,

I would like to know how to write script for browser enhancements such as this.....

I remember your last full Noia theme addon package back in 2014...you provided for Firefox before the platform was altered & you stated it was too time consuming to keep updates coming one after another because the browser kept altering, it was sad to hear that.

I've been trying to get the current version of Firefox to appear as Pale Moon does with Noia.

Now it's close, the tabs & the tab bar still need to shrink the height down to 20 (what do you think)?

Being on a laptop space is at a premium.

I still have your xpi addon package for Noia 4 1.8.9 sitting unused in my Pale Moon browser

Tony

     On Sunday, October 29, 2023 at 01:19:05 p.m. EDT, Aris ***@***.***> wrote:  
 
 


Give 3.0.1 release a try: https://github.com/Aris-t2/NoiaButtons/releases

**Reply by Aris-t2** [reply]
Doing this for over 12 years makes you know where to look for most of the stuff. Firefox also offers a good way to inspect ui content, so looking into code is possible most of the time.

Reducing tabs height might be a bit tricky here, because it may cause glitches and overlapping.

Try to play with these values, after adding the code to the bottom of your userChrome.css:

```css
:root {
  --tab-min-height: 22px !important;
}

#main-window:not([chromehidden*="toolbar"])[uidensity="compact"] #navigator-toolbox {
  padding-bottom: calc(2px + var(--tab-min-height)) !important;
}

#main-window:not([chromehidden*="toolbar"]):not([uidensity="compact"]):not([uidensity="touch"]) #navigator-toolbox {
  padding-bottom: calc(7px + var(--tab-min-height)) !important;
}

#main-window:not([chromehidden*="toolbar"])[uidensity="touch"] #navigator-toolbox {
  padding-bottom: calc(7px + var(--tab-min-height)) !important;
}

#TabsToolbar {
  height: calc(7px + var(--tab-min-height)) !important;
}
```

Thes two were active during testing:
```
noiaui_light_tabs_below_navigation_toolbar.css
noiaui_light_separators_between_toolbars.css
```

**Comment by 55trucker** [comment]
Here's what your work has produced....the compact mode is enabled, the horizontal separators are visible, the tabs bar has shrunk to 22, *an entry placed into the about:config to enter the vertical spacers is there........

the only item that as far as I understand cannot be introduced which isn't present is the status bar at the bottom of the screen (which I use).

Also in the userchrome file is an *import* pointer for the drop down bookmarks customization to alter the basic folder icons to W10 folder icons

I sent along the userchrome.css file which has all of this in it....

is there anything else you can think where tweaking for Noia is concerned?


Tony 


     On Monday, October 30, 2023 at 04:46:20 a.m. EDT, Aris ***@***.***> wrote:  
 
 


Doing this for over 12 years makes you know where to look for most of the stuff. Firefox also offers a good way to inspect ui content, so looking into code is possible most of the time.

Reducing tabs height might be a bit tricky here, because it may cause glitches and overlapping.

Try to play with these values, after adding the code to the bottom of your userChrome.css:
:root {  --tab-min-height: 22px !important;}#main-window:not([chromehidden*="toolbar"])[uidensity="compact"] #navigator-toolbox {  padding-bottom: calc(2px + var(--tab-min-height)) !important;}#main-window:not([chromehidden*="toolbar"]):not([uidensity="compact"]):not([uidensity="touch"]) #navigator-toolbox {  padding-bottom: calc(7px + var(--tab-min-height)) !important;}#main-window:not([chromehidden*="toolbar"])[uidensity="touch"] #navigator-toolbox {  padding-bottom: calc(7px + var(--tab-min-height)) !important;}#TabsToolbar {  height: calc(7px + var(--tab-min-height)) !important;}
Thes two were active during testing:
noiaui_light_tabs_below_navigation_toolbar.cssnoiaui_light_separators_between_toolbars.css

**Comment by 55trucker** [comment]
Aris:

what are the chances that your work might also work in Vivaldi?
Vivaldi is a chrome browser, would the script be compatible?

**Reply by pro100Hub** [reply]
@Aris-t2 
Upgraded to FF 135 and now the `titlebar-buttonbox-container` is at the very bottom.  Its permanent place should be in the top right corner.
Maybe the code `noiaui_light_tabs_below_navigation_toolbar.css` I'm using is outdated.

![firefox_6wf2iD5yGN](https://github.com/user-attachments/assets/43e0d5cf-63b7-4c8e-bfd8-9368833776cf)

**Comment by 55trucker** [comment]
@pro100Hub: 
I'd turn off the browser auto update (If it is active), roll  back to 133.03 & continue to use that version.......
I'm also using Aris's Noia Buttons & tabs below nav bar on 133.03.....runs flawlessly .......I've also tweaked the browser so that it looks just like the old entire Noia theme

**Comment by Kaylikon** [comment]
Hello,
How can I change the color of the Firefox button?
Note: I'm a complete novice at this, so thank you for your help.
![fırefox](https://github.com/user-attachments/assets/377b577f-89ee-4cae-9730-e425992f9fad)
